test/concat_test.dart

```diff
@@ -8,40 +8,121 @@ import 'package:test/test.dart';
 import 'package:stream_transform/stream_transform.dart';
 
 void main() {
-  group('concat', () {
-    test('adds all values from both streams', () async {
-      var first = new Stream.fromIterable([1, 2, 3]);
-      var second = new Stream.fromIterable([4, 5, 6]);
-      var all = await first.transform(concat(second)).toList();
-      expect(all, [1, 2, 3, 4, 5, 6]);
-    });
-
-    test('closes first stream on cancel', () async {
-      var firstStreamClosed = false;
-      var first = new StreamController()
-        ..onCancel = () {
-          firstStreamClosed = true;
-        };
-      var second = new StreamController();
-      var subscription =
-          first.stream.transform(concat(second.stream)).listen((_) {});
-      await subscription.cancel();
-      expect(firstStreamClosed, true);
-    });
-
-    test('closes second stream on cancel if first stream done', () async {
-      var first = new StreamController();
-      var secondStreamClosed = false;
-      var second = new StreamController()
-        ..onCancel = () {
-          secondStreamClosed = true;
-        };
-      var subscription =
-          first.stream.transform(concat(second.stream)).listen((_) {});
-      await first.close();
-      await new Future(() {});
-      await subscription.cancel();
-      expect(secondStreamClosed, true);
-    });
-  });
+  var streamTypes = {
+    'single subscription': () => new StreamController(),
+    'broadcast': () => new StreamController.broadcast()
+  };
+  for (var firstType in streamTypes.keys) {
+    for (var secondType in streamTypes.keys) {
+      group('concat [$firstType] with [$secondType]', () {
+        StreamController first;
+        StreamController second;
+
+        List emittedValues;
+        bool firstCanceled;
+        bool secondCanceled;
+        bool secondListened;
+        bool isDone;
+        List errors;
+        Stream transformed;
+        StreamSubscription subscription;
+
+        setUp(() async {
+          firstCanceled = false;
+          secondCanceled = false;
+          secondListened = false;
+          first = streamTypes[firstType]()
+            ..onCancel = () {
+              firstCanceled = true;
+            };
```

Comment:
I didn't think it was critical since the test that values come through prove we've listened when we should.